src/Symfony/Bridge/Twig/Extension/SecurityExtension.php

```diff
@@ -14,21 +14,25 @@
 use Symfony\Component\Security\Acl\Voter\FieldVote;
 use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
 use Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException;
+use Symfony\Component\Security\Http\Logout\ImpersonateUrlGenerator;
 use Twig\Extension\AbstractExtension;
 use Twig\TwigFunction;
 
 /**
  * SecurityExtension exposes security context features.
  *
  * @author Fabien Potencier <fabien@symfony.com>
+ * @author Amrouche Hamza <hamza.simperfit@gmail.com>
  */
 class SecurityExtension extends AbstractExtension
 {
     private $securityChecker;
+    private $impersonateUrlGenerator;
 
-    public function __construct(AuthorizationCheckerInterface $securityChecker = null)
+    public function __construct(AuthorizationCheckerInterface $securityChecker = null, ImpersonateUrlGenerator $impersonateUrlGenerator)
```

Comment:
I think PHP strongly discourages non-optional arguments after optional

http://php.net/manual/en/functions.arguments.php
> Note that when using default arguments, any defaults should be on the right side of any non-default arguments; otherwise, things will not work as expected.